shell/common/shell_test_platform_view.cc

```diff
@@ -3,7 +3,13 @@
 // found in the LICENSE file.
 
 #include "flutter/shell/common/shell_test_platform_view.h"
+
+#ifdef SHELL_ENABLE_GL
```

Comment:
Even when using Vulkan, we never have to link with libVulkan.so. So we will never get into a situation where where the test harness fails to link if Vulkan is not used. The exact same logic applies for OpenGL as well. As written, we will need two variants of the test runner for OpenGL and Vulkan. I suggest we have just one which is both OpenGL and Vulkan capable. At runtime, if the test or fixture asks use to initialize an unsupported client rendering API, we can skip the test or trip an assertion.

This way, all platforms will be Vulkan capable and their ability to actually exercise the Vulkan backend will depend on whether that platform can provide a libVulkan.so for proc resolution.

While we will always use the libVulkan.so from SwiftShader, this scheme should give us the flexibility to run with an alternative Vulkan implemenation.

Reply:
Spoke offline about this - tried this but due to compilation issues this can't be done easily right now, and it's not worth blocking enabling these tests for this. Filed https://github.com/flutter/flutter/issues/50298 detailing the problems.